pkg/sql/compile/dml.go

```diff
@@ -105,6 +105,24 @@ func fillBatch(bat *batch.Batch, p *plan.InsertValues, rows []tree.Exprs, proc *
 
 	for i, v := range bat.Vecs {
 		switch v.Typ.Oid {
+		case types.T_uuid:
+			vs := make([]types.Uuid, rowCount)
+			{
+				for j, expr := range p.Columns[i].Column {
+					vec, err := colexec.EvalExpr(tmpBat, proc, expr)
+					if err != nil {
+						return y.MakeInsertError(v.Typ.Oid, p.ExplicitCols[i], rows, i, j)
+					}
+					if nulls.Any(vec.Nsp) {
+						nulls.Add(v.Nsp, uint64(j))
+					} else {
+						vs[j] = vec.Col.([]types.Uuid)[0]
+					}
```

Comment:
This is strange code, what is j here?     vs is size rowCount, but here we loop over p.Columns.Column?

Reply:
Let me read the original code and answer this question. Let's take int8 data as an example. The code is as follows:
/*
bat *batch.Batch: The Batch structure of the data to be inserted, which contains the Vector array, 
and each Vector corresponds to each column of data to be inserted.
p *plan.InsertValues：The multi row tuple data inserted by the insert statement corresponds to the data structure in plan2.
rows []tree.Exprs： The multi line tuple data inserted by the insert statement corresponds to the data structure 
in the ast syntax tree.
*/
func fillBatch(bat *batch.Batch, p *plan.InsertValues, rows []tree.Exprs, proc *process.Process) error {

	/* 
	The data inserted by insert is a two-dimensional table structure, so p.columns [i] corresponds to the data in the ith column. 
	Each column contains the same number of rows, so len (p.columns [0]. Column) is the same
	*/
	
	// Get the number of rows contained in column 0. In fact, rowcount is the number of rows of the tuple to be inserted
	rowCount := len(p.Columns[0].Column)
	tmpBat := batch.NewWithSize(0)
	tmpBat.Zs = []int64{1}

	// Use for range to traverse bat.vecs. I corresponds to the number of each column, and V corresponds to the data of each column
	for i, v := range bat.Vecs {
		switch v.Typ.Oid {
		case types.T_int8:
			vs := make([]int8, rowCount)
			{
				// Traverse and calculate the expressions of all rows in the ith column, and j is used to index the number of rows
				for j, expr := range p.Columns[i].Column {
				    // Calculate the data value of the jth row of the ith column and return a single value
					vec, err := colexec.EvalExpr(tmpBat, proc, expr)
					if err != nil {
						return y.MakeInsertError(v.Typ.Oid, p.ExplicitCols[i], rows, i, j)
					}
					if nulls.Any(vec.Nsp) {
						nulls.Add(v.Nsp, uint64(j))
					} else {
					    // Put the calculation result in the jth row of the current column vector
						vs[j] = vec.Col.([]int8)[0]
					}
				}
			}
			if err := vector.AppendFixed(v, vs, proc.Mp()); err != nil {
				return err
			}
		}
	}
}